gcp/opentelemetry-demo-workload-identity.yaml.gotmpl

```diff
@@ -23,3 +23,57 @@ opentelemetry-collector:
     annotations:
       "iam.gke.io/gcp-service-account": "opentelemetry-demo@{{ . }}.iam.gserviceaccount.com"
 {{ end }}
+
+{{ if .Values.collector_image_repository }}
+  image:
+{{ with .Values | getOrNil "collector_image_repository" }}
+    repository: "{{ . }}"
+{{ end }}
+{{ if .Values.collector_image_tag }}
+{{ with .Values | getOrNil "collector_image_tag" }}
+    tag: "{{ . }}"
+{{ end }}
+{{ end}}
+{{ end}}
+
+{{ if .Values.otlp_endpoint }}
+  config:
+    exporters:
+      otlphttp:
```

Comment:
The new exporter and processor should probably have explicit names so they don't conflict with anything the upstream demo happens to configure. e.g. the upstream demo already has an `otlp` exporter and an `otlphttp/prometheus` exporter.

Also, why is this an HTTP exporter and not a gRPC exporter? We should always prefer gRPC for talking to Google APIs.

Reply:
Yeah, I wasn't sure what to do for the names. It's fine for the exporter, but if we give the processor a new name then we have to keep the list of processors in the pipeline in sync by re-listing each one. That's probably fine though.

For http vs grpc, is there a reason to use grpc? The endpoint supports both, but I'm not sure which one is more preferred by otel

Reply:
@quentinmit trying to switch to the gRPC exporter it seems like there's an issue with it reaching the otlp endpoint. a couple people have seen that, and giving it a day of debugging I haven't been able to fix it. So, I suggest we go with the http exporter for now (because it works) and track down why gRPC is broken in a separate bug

Reply:
That's fine in this PR. gRPC has better performance, especially when sending lots of small RPCs, so to the extent this repo serves as a best practices guide, we should try to get gRPC working and switch to it.